posts/2021/12/conda-recipe-grayskull.md

```diff
@@ -0,0 +1,118 @@
+<!--
+.. title: On Software Packages, Conda and Recipes
+.. slug: conda-recipe-grayskull
+.. date: 2021-12-08 10:00:00 UTC+00:00
+.. author: Mahe Iram Khan
+.. tags: Conda, conda-forge, recipes, Grayskull, packaging, Python
+.. category:
+.. link:
+.. description:
+.. type: text
+.. previewimage: /images/2021/12/grayskull.jpg
+-->
+
+Python might be the most popular snake out there, but most of us have also heard of that other serpent: Conda. And some of us have wondered what it really is. In this post we’ll learn about Conda, software packages and package recipes. Most importantly we’ll learn about Grayskull — a conda recipe generator.
+
+<!-- TEASER_END -->
+
+Hey! I’m [Mahe](https://twitter.com/IramMahe), a Computer Engineering student from India.
+During the summer of 2021 I was an intern at Quansight Labs and I worked on a project called ‘Grayskull’.
+Before we learn about Grayskull, as promised, I’ll talk about software packages and Conda.
+
+## Software Packages
+A software package is, simply put, a working piece of code that somebody wrote and published for others to use. A functional packet of code that does something — a package. You install packages for your use. Sometimes you import them into your code while writing a package of your own.
+
+```
+import numpy
+```
+That’s you ‘importing’ the package numpy into your code.
+
+## Channels
+Once you write a package you might want to publish it so that others can download it and use it. Packages are published on ‘channels’.
+Channels are like warehouses of packages.
+
+## Conda
+Conda is an OS-agnostic package manager with great popularity in the Python world and data science adjacent libraries.
+
+Conda-build is a set of commands and tools that lets you build your own packages for Conda. These tools let you manage the environments and dependencies of your packages and generate the needed context for your project.
+
+Anaconda provides a default channel called ‘defaults’ where packages are published. There are several community driven channels as well, conda-forge being the most popular one.
+From here onwards in this blog we’ll assume that we want to publish our package on the conda-forge channel.
+
+## Publishing a package on conda-forge
+Publishing packages on the conda-forge channel requires the knowledge of ‘recipes’. A recipe is a collection of files that defines how to build a package. Minimally a recipe contains a meta.yaml file that describes:
+- The package name and version
+- its dependencies
+- how to build it
+- some other metadata
+You can learn more about recipes [here](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#meta-yaml).
+
+<p align="center">
+    <img
+     alt="Cute representation of a package recipe"
+     src="/images/2021/12/recipe_animation.png">
+    <i><br>Cute representation of a package recipe</i>
+</p>
```

Comment:
I have moved this to the conda-build section, so we focus on conda-forge here.

```suggestion
Since conda-forge relies on Conda-build, publishing packages on the conda-forge channel requires creating recipes.
```